### PR TITLE
RUM-1823: Add buildId to the RUM error and Log events

### PR DESCRIPTION
**Pull request by 0xnm**
### What does this PR do?

This PR adds `buildId` support for RUM error events and Log events.

`buildId` will be generated by the Datadog Gradle Plugin (see https://github.com/DataDog/dd-sdk-android-gradle-plugin/pull/211) only for builds which are minified.

Once attached to RUM error events and Logs events, it will be used for the symbolication.

**NB**: although `build_id` is declared in the common schema for RUM events, this PR adds it only for error events.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt [hunk 1]
```diff
@@ -332,6 +334,21 @@ internal class CoreFeature(
         serviceName = configuration.service ?: appContext.packageName
         envName = configuration.env
         variant = configuration.variant
+        appBuildId = with(appContext.assets) {
+            try {
+                open(BUILD_ID_FILE_NAME).bufferedReader().use {
+                    it.readText().trim()
+                }
+            } catch (@Suppress("SwallowedException") e: IOException) {
```
Comment (xgouchet) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt:342]: > [!NOTE]
> We might want to distinguish `FileNotFoundException` from other kinds of `IOException`
> The first might mean that the app was built with an older plugin (or no plugin at all) in which case 
> it's expected to not have a Build ID File. In that case an `INFO` log is indeed enough.
> The second means an actual IO error happened with a file that exists, and this should report at least to `WARNING` or `ERROR` (and maybe be sent to telemetry).

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/CoreFeatureTest.kt [hunk 2]
```diff
@@ -464,6 +473,44 @@ internal class CoreFeatureTest {
...
+        // Then
+        assertThat(testedFeature.appBuildId).isEqualTo(fakeBuildId.toString())
+    }
+
+    @Test
+    fun `𝕄 initializes build ID 𝕎 initialize() { build ID is missing }`() {
+        // Given
+        whenever(
+            appContext.mockInstance.assets.open(CoreFeature.BUILD_ID_FILE_NAME)
+        ) doThrow IOException()
```
Comment (xgouchet) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/CoreFeatureTest.kt:495]: > [!NOTE]
> This should be a `FileNotFoundException` to properly mock an asset not being present.

Comment (xgouchet) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/CoreFeatureTest.kt:495]: You can add other tests for other scenarios: 
- `RuntimeException` could be thrown if the `AssetManager` has been closed (cf: `AssetManager.ensureOpenLocked()`
- `IOException` if the `read()` operation fails

**Review by mariusc83 — APPROVED** [review]

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt (hunk 1 above)
Comment (0xnm) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt]: Indeed, good point. Makes sense to distinguish between the case when file is not there and when something went wrong during the file read.

**Review by xgouchet — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**